fun/connect4System/reactionListener.go.tmpl

```diff
@@ -0,0 +1,194 @@
+        {{- range $board -}}
+                {{$rows := ""}}
+                {{- range . -}}
+                        {{- $rows = printf "%s%s " $rows (str . | $store.emojis.Get) -}}
+                {{- end -}}
+                {{- $out = print $rows "\n" $out -}}
+        {{- end -}}
+        {{- $out = print $out "1️⃣ 2️⃣ 3️⃣ 4️⃣ 5️⃣ 6️⃣ 7️⃣" -}}
+        {{- $embed := sdict "author" (sdict "name" (print .cPlayer.Username "'s turn") "icon_url" (.cPlayer.AvatarURL "256")) 
+            "title" "Connect4" 
+            "description" $out
+            "color" (str .turn | $store.color.Get)
+            "fields" (cslice 1 2)
+            "footer" (sdict "text" "Powered by • Yagpdb.xyz" ) -}}
+        {{- $embed.fields.Set (sub .turn 1) (sdict "name" (print "Player " .turn) "value" (print "> " .cPlayer.Mention) "inline" true) -}}
+        {{- $embed.fields.Set (sub .nextTurn 1 ) (sdict "name" (print "Player " .nextTurn) "value" .nPlayer.Mention "inline" true ) -}}
+        {{- .Set "embed" $embed -}}
+{{- end}}
+
+{{/* Global variables */}}
+{{ $data := sdict}}{{$players := cslice}}{{$cPlayer := ""}}{{$turn := ""}}{{$nextTurn := ""}}{{$nPlayer := ""}}{{$input := ""}}{{$flag := false}}
+{{ $store := sdict
+    "emojis" (sdict "0" "⚫" "1" "🔴" "2" "🟡" "10" "🔵") 
+    "color" (sdict "1" 0xff4d12 "2" 0xfff457)}}
+
+{{ if $db := dbGet 2021 "connect4" }}
+    {{ $data = sdict $db.Value}}
+    {{ $players = $players.AppendSlice $data.players}}{{/* slice ["offset",p1,p2] */}}
+    {{ $turn = $data.turn|toInt}} {{/* 1 or 2 */}}
+    {{ $cPlayer = (index $players $turn|toInt64|getMember).User}}
+    {{ $nextTurn = sub 3 $turn}} {{/* switch between 1 and 2 */}}
+    {{ $nPlayer = (index $players $nextTurn|toInt64|getMember).User}}
+
+    {{ if eq ($data.msgID|toInt64) .Message.ID}}
+            {{ deleteMessageReaction nil .Message.ID .User.ID .Reaction.Emoji.APIName}}
+            {{/* quit reaction */}}
+            {{ if and (eq .Reaction.Emoji.ID 844556617085485058) (in $players (str .User.ID)) }}
+                    {{ $otherPlayer := or (and (eq $cPlayer.ID .User.ID) $nPlayer) $cPlayer}}
+                    {{ $embed := index .Message.Embeds 0 | structToSdict}}
+                    {{ $embed.Set "author" (sdict "name" "Game Over" "icon_url" ($otherPlayer.AvatarURL "256"))}}
+                    {{ $embed.Set "color" 0xffdc42}}
+                    {{ editMessage nil .Message.ID (complexMessageEdit "content" (printf "> <a:r_leave:844556617085485058>┃**%s** have left the game\n🎊┃**Winner: %s**" .User.Username $otherPlayer.Mention) "embed" (cembed $embed))}}
+                    {{ deleteAllMessageReactions nil .Message.ID }}
+                    {{ dbDel 2021 "connect4"}}
+            {{ else }}
+                    {{ if eq $cPlayer.ID $.User.ID}}
+                            {{/* Getting integer from keycap reaction */}}
+                            {{ if and ($temp := (index (toRune .Reaction.Emoji.Name) 0|printf "%c"|toInt)) (le $temp 7)}}
```

Comment:
```suggestion
                            {{ if le ($temp := index (toRune .Reaction.Emoji.Name) 0) '7'}}
```
check if this works first though

Reply:
it does and wow, I did not know u could use single quotes that way.
 Thanks for suggesting it ^ ^